### ISSUE TITLE
Testcafe throwing error Error in Role initializer

### ISSUE THREAD

**Issue by damorton** [DevExpress/testcafe#2473]
### Are you requesting a feature or reporting a bug?
Bug

### What is the current behavior?
testcafe throwing error

### What is the expected behavior?
testcafe does not throw error...

#### Provide the test code and the tested page URL (if applicable)

```
import { Role, Selector } from 'testcafe';

const loginPageUrl = 'http://localhost:8888/#/login';

const login = () =>
  Role(loginPageUrl, async (controller) => {
    await controller
      .typeText(Selector('input').withAttribute('data-testid', 'username'), 'dev')
      .typeText(Selector('input').withAttribute('data-testid', 'password'), 'password')
      .click(Selector('button').withAttribute('data-testid', 'navigation-login-link'));
  });

const devuser = login();

fixture('Test').page('http://localhost:8888/#/');

test('test', async (controller) => {
  await controller
    // login
    .useRole(devuser)
    // navigate to login page
    .navigateTo(loginPageUrl);
});

```
### Specify your

* macOS
* node v8
* testcafe 0.20.2

### Error 

```
Running tests in:
 - Chrome 67.0.3396 / Mac OS X 10.13.4

 Test
 ✖ test

   1) - Error in Role initializer -
      Error on page "http://localhost:8888/#/":

      Uncaught TypeError: Cannot read property 'onBodyElementMutation' of
      undefined

      Browser: Chrome 67.0.3396 / Mac OS X 10.13.4
```

**Comment by damorton** [comment]
New test case without `Role`

```
import { Selector } from 'testcafe';

const loginPageUrl = 'http://localhost:8888/#/login';

fixture('Test').page('http://localhost:8888/#/');
test('test', async (controller) => {
  await controller
    .navigateTo(loginPageUrl)
    .typeText(Selector('input').withAttribute('data-testid', 'username'), 'dev')
    .typeText(Selector('input').withAttribute('data-testid', 'password'), 'password')
    .click(Selector('button').withAttribute('data-testid', 'navigation-login-link'))
    .navigateTo(loginPageUrl);
});

```

Getting error 

```
Running tests in:
 - Chrome 67.0.3396 / Mac OS X 10.13.4

 Test
 ✖ test

   1) Error on page "http://localhost:8888/#/":

      Uncaught TypeError: Cannot read property 'onBodyElementMutation' of
      undefined

      Browser: Chrome 67.0.3396 / Mac OS X 10.13.4

          3 |const loginPageUrl = 'http://localhost:8888/#/login';
          4 |
          5 |fixture('Test').page('http://localhost:8888/#/');
          6 |test('test', async (controller) => {
          7 |  await controller
       >  8 |    .navigateTo(loginPageUrl)
          9 |    .typeText(Selector('input').withAttribute('data-testid',
      'username'), 'dev')
         10 |    .typeText(Selector('input').withAttribute('data-testid',
      'password'), 'password')
         11 |    .click(Selector('button').withAttribute('data-testid',
      'navigation-login-link'))
         12 |    .navigateTo(loginPageUrl);
         13 |});

```

**Comment by damorton** [comment]
Another test in Firefox

```
import { Role, Selector } from 'testcafe';

const baseUrl = 'http://localhost:8888/';
const login = () =>
  Role(`${baseUrl}#/login`, async (controller) => {
    await controller
      .typeText(Selector('input').withAttribute('data-testid', 'username'), 'dev')
      .typeText(Selector('input').withAttribute('data-testid', 'password'), 'password')
      .click(Selector('button').withAttribute('data-testid', 'login-button'));
  });

// start on base url
fixture('Test').page(baseUrl);

// navigate to login and authenticate
const devuser = login();

test('test', async (controller) => {
  await controller.useRole(devuser).navigateTo(`${baseUrl}#/about`);
});

```

Error 

```
Running tests in:
 - Firefox 52.0.0 / Mac OS X 10.13.0

 Test
 ✖ test

   1) - Error in Role initializer -
      Error on page "http://localhost:8888/#/":

      TypeError: t.shadowUI is undefined

      Browser: Firefox 52.0.0 / Mac OS X 10.13.0
```

**Comment by AndreyBelym** [comment]
Hi @damorton, have you found the reason of the issue?

**Comment by vzaidman** [comment]
I have
```
Uncaught TypeError: Cannot read property 'onBodyElementMutation' of undefined
```
too. on a testcase that worked on 0.19 but i cant easily add a testcase here

**Comment by AndreyBelym** [comment]
@vzaidman, I think we have found the reason of the issue - https://github.com/DevExpress/testcafe-hammerhead/issues/1639. @LavrovArtem is working on a fix now.

**Comment by damorton** [comment]
Closed this because I thought it was my project configuration, couldn't reproduce in a new CRA project. Rolled back to 0.19.2.

**Comment by TestAndunR** [comment]
Have you guys fix this issue, Its available in  0.21.1 as well
